### PR TITLE
pythonPackages.tomli: init at 1.0.4

### PR DESCRIPTION
**Pull request by veehaitch**
###### Motivation for this change

Also, soon to be required by the next `mypy` release: https://github.com/python/mypy/pull/10824

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130581` at bb4f5836 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.tomli</li>
    <li>python39Packages.tomli</li>
  </ul>
</details>

File: pkgs/development/python-modules/tomli/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchFromGitHub {
+    owner = "hukkin";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-ld0PsYnxVH3RbLG/NpvLDj9UhAe+QgwCQVXgGgqh8kE=";
+  };
+
+  nativeBuildInputs = [
+    flit-core
+    pytestCheckHook
```
Comment (mweinelt) [inline pkgs/development/python-modules/tomli/default.nix:23]: The `pytestCheckHook` should be in `checkInputs`.

Comment (veehaitch) [reply pkgs/development/python-modules/tomli/default.nix:24]: Thanks, fixed!

**Comment by evils** [comment]
seems to work
about half the time of native, almost 10x the time of rtoml in my [crude benchmark](https://gitlab.com/evils/py-dict-file-bench) (large file) though

thanks

**Review by mweinelt — APPROVED** [review]

**Pull request merged**